Results
===

![Image](https://github.com/pilarcormo/Correlations/blob/master/Results/Rplot.700.2000contigs.png?raw=false)

The increase in the number of divisions decrease the correlation between the ratios in all the cases under study. 

When we progressively increase the size of the genome, the correlation is good at high number of genome divisions. Example: in the first graph, the correlation between the 1 Mb genome when using 100 divisions is lower than 0.5 but when the genome size is increased to 3 or 5 Mb, the correlation is close to 1. This increase is true until 3 Mb genomes in all the cases. 

I am tented to say that the correlation is in general better when we have a higher number of heterozygous SNP (1/500). I kept the number of homozygous SNP more or less constant for all the genome sizes. 

In the bottom graphs, we see a clear improvement in the larger genomes when using a higher number of contigs. This is logical, because there seems to be an optimal contig size to calculate the ht/hm ratio effectively. 

In contrast, in the upper-part graphs, there is an improvement in the correlation for the 7-10 Mb genomes when increasing the number of contains. 



#####Try now: 

- 1/1500 and 1/2000 ht SNPs

- Change the hm SNP distribution too

- As in the 3rd graph, try to build the plots with more divisions.
